### ISSUE TITLE
Option for RMS and VPL limiting

### ISSUE THREAD

**Issue by hgetta** [HEnquist/camilladsp#307]
I am using camilladsp in a maybe slightly unconventional fashion for live PA. Currently I am experimenting with the compressor in v2 but would it be possible to add premade limiter options? Would also be great for diy sub builds etc.

**Comment by HEnquist** [comment]
Do you mean some presets for the compressor and limiter? Could be done, then I would probably implement it as a feature of the gui.
When you add a compressor in the gui it adds one with a very general configuration. Is it something like that you mean, but with more presets to choose from? What would the additional presets be?

**Comment by hgetta** [comment]
Thanks for your fast reply!

I think presets (i guess like the biquad hp lp filters?) would work but if you could implement ui elements (with translation to the compressor settings in the background) similar to an online calculator like this https://leaprofessional.com/loudspeaker-limiter-calculator/ I guess it would be more easy to understand and implement. 


Am 05.12.2023 um 21:44 schrieb Henrik Enquist ***@***.***>:

﻿

Do you mean some presets for the compressor and limiter? Could be done, then I would probably implement it as a feature of the gui.
When you add a compressor in the gui it adds one with a very general configuration. Is it something like that you mean, but with more presets to choose from? What would the additional presets be?

**Comment by HEnquist** [comment]
Ok thanks, now I get it. The sheet is unfortunately protected, so it's not possible to see the formulas it uses to calculate the compressor parameters. The protection can probably be removed, but it wouldn't help as I can't use formulas acquired that way.
Anyway, the sheet seems to give quite general recommendations for the compressor settings. Those are probably more useful as a starting point than a final result.
I found some more here: https://www.powersoft.com/wp-content/uploads/2019/01/powersoft_TN009_LimiterSetup_en_v1.0.pdf
It's even more generic, but this is at least something I could probably use. Maybe as some kind of wizard in the gui, where you make choices that are then used to generate parameters as a starting point for a compressor.

**Comment by mike811** [comment]
Hello,
since impedance does vary a lot in the bass region like displayed in figure 6 of the above mentioned PDF, wouldn’t a real impedance measurement (for subwoofers) be useful in this context? At least for DIYers? I'm having some lookup table in mind where Camilladsp could calculate the current flowing based on signal voltage and real impedance in frequency bands and not only with a fixed 8 or 4 ohm value.

**Comment by HEnquist** [comment]
> Hello, since impedance does vary a lot in the bass region like displayed in figure 6 of the above mentioned PDF, wouldn’t a real impedance measurement (for subwoofers) be useful in this context? At least for DIYers? I'm having some lookup table in mind where Camilladsp could calculate the current flowing based on signal voltage and real impedance in frequency bands and not only with a fixed 8 or 4 ohm value.

Yes that would probably work quite well, but this would be a substantial project in itself. It's also well outside the scope I have in mind for camilladsp.

**Comment by HEnquist** [comment]
After thinking some more about this, I'm not sure there is enough to gain from implementing the calculations inside the camilladsp gui versus just using an excel sheet or some online calculator. 
Implementing new features in the gui is quite time consuming, and free time is limited so I have to spend it wisely.

**Comment by HEnquist** [comment]
Ok, a wizard should be doable, I'll put it on the list. With some luck some of the plumbing from the upcoming config import functionality can be reused for this. 

I just work on this because I enjoy it and I don't ask for or expect any donations. So I don't have any fancy donation link, but if you really really want, then PayPal works (my email is in my profile).
